### DISCUSSION TITLE
callback_notify configuration has been removed from run()

### DISCUSSION THREAD

**Discussion by BlueMagma2** [Kludex/uvicorn#1712] [Potential Issue] [answered]
There is a `callback_notify` parameter available in `Config`: https://github.com/encode/uvicorn/blob/0.18.3/uvicorn/config.py#L248

up until 0.17.6, `uvicorn.run(...)` took  `**kwargs` and supported all parameters that were supported by Config: https://github.com/encode/uvicorn/blob/0.17.6/uvicorn/main.py#L440

Starting in 0.18.0, `uvicorn.run(...)` is restricting what parameters are available by listing what them, but even though `callback_notify` is still available in `Config`, it is not made available when calling `uvicorn.rrun()`: https://github.com/encode/uvicorn/blob/0.18.0/uvicorn/main.py#L457

I suggest one of these solutions:
- Add `callback_notify` (and other missing parameters if any) in `uvicorn.run()`
- Remove `callback_notify` from `Config`
- Allow caller to pass their own `Config` object to `uvicorn.run()`

**Comment by Kludex** [comment, marked as answer]
`callback_notify` was created for the `UvicornWorker`, that's why it was not included in the `run()`.

Reference: https://github.com/encode/uvicorn/pull/1545#discussion_r912048384
